### PR TITLE
Proposer: better handling of blobs bundle

### PR DESCRIPTION
**Pull request by terencechain**
### Summary
This PR refactors how `blindBlobsBundle` and `fullBlobsBundle` are managed, transitioning them from function parameters to package-level variables. Additionally, it introduces new helper methods that embed `setKzgCommitments` logic, improving the cohesiveness of payload-related operations.

### Background
Previously, `blindBlobsBundle` and `fullBlobsBundle` were passed around as parameters in multiple functions. This led to unwieldy function signatures and decreased code readability. Separately, the `setKzgCommitments` functionality was distanced from where the payload was set, creating potential issues for future development.

### Changes
1. Move `blindBlobsBundle` and `fullBlobsBundle` to package-level variables.
    - Reduces the number of parameters in several functions like `BuildBlockParallel`.
   
2. Introduce new helper methods `setExecution`, `setLocalExecution`, and `setBuilderExecution`.
    - These methods internally call `setKzgCommitments`, ensuring that it is invoked whenever a payload is set.

3. Refactor `setExecutionData` to use the new helper methods based on the execution context.

### CODE REVIEW

File: beacon-chain/rpc/prysm/v1alpha1/validator/proposer_execution_payload.go
Comment (james-prysm) [inline beacon-chain/rpc/prysm/v1alpha1/validator/proposer_execution_payload.go:89]: do we not need to use the fullBlobsBundle variable here? does that variable need to be set?

File: beacon-chain/rpc/prysm/v1alpha1/validator/proposer.go
Comment (james-prysm) [inline beacon-chain/rpc/prysm/v1alpha1/validator/proposer.go:52]: will it be possible for fullBlobsBundle or blindBlobsBundle to be misaligned to their associated block somehow?

Comment (terencechain) [reply beacon-chain/rpc/prysm/v1alpha1/validator/proposer.go:52]: This is a good point. To prevent leakover, I think we set the variable to nil after it's used.

Comment (terencechain) [reply beacon-chain/rpc/prysm/v1alpha1/validator/proposer.go:52]: [ee64a3b](https://github.com/prysmaticlabs/prysm/pull/12956/commits/ee64a3b864d74e2cc1661ddede7fcf49f06d5375)

File: beacon-chain/rpc/prysm/v1alpha1/validator/proposer.go
Comment (james-prysm) [inline beacon-chain/rpc/prysm/v1alpha1/validator/proposer.go:185]: should we also set it to nil if it errors using it above?

Comment (terencechain) [reply beacon-chain/rpc/prysm/v1alpha1/validator/proposer.go]: yep 26a93c52c

**Review by james-prysm — APPROVED** [review]
lgtm felt my concerns were addressed. feel free to get another pair of eyes to review before merging.

**Comment by terencechain** [comment]
Local interop passes for me for full blobs

File: beacon-chain/rpc/prysm/v1alpha1/validator/proposer.go
Comment (saolyn) [inline beacon-chain/rpc/prysm/v1alpha1/validator/proposer.go:161]: I am a bit confused by this function. It seems to only return the payload and within `getBuilderPayloadAndBlobs` we don't seem to be doing anything with `fullBlobsBundle`. Am I missing something?

Comment (terencechain) [reply beacon-chain/rpc/prysm/v1alpha1/validator/proposer.go:161]: `fullBlobsBundle` is cached in `getLocalPayloadAndBlobs`

File: beacon-chain/rpc/prysm/v1alpha1/validator/proposer_bellatrix.go
Comment (rkapka) [inline beacon-chain/rpc/prysm/v1alpha1/validator/proposer_bellatrix.go:298]:
```suggestion
	if blindBlobsBundle != nil {
```

**Review by rkapka — CHANGES_REQUESTED** [review]

**Review by rkapka — APPROVED** [review]

**Review by james-prysm — APPROVED** [review]

**Pull request merged**
